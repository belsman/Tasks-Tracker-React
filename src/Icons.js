import React from 'react';
import { faHourglass, faPlus, faChartLine, faChartPie, faEllipsisH, faAngleRight } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export const AddMeasurementIcon = () => (
  <span>
    <FontAwesomeIcon icon={faHourglass} size="lg"/>
    <sup><FontAwesomeIcon icon={faPlus} size="sm"/></sup>
  </span>
);

export const ListMeasurementIcon = () => (
    <span>
      <FontAwesomeIcon icon={faChartLine} size="lg"/>
    </span>
);

export const ProgressIcon = () => (
    <span>
      <FontAwesomeIcon icon={faChartPie} size="lg"/>
    </span>
);

export const MoreIcon = () => (
    <span>
      <FontAwesomeIcon icon={faEllipsisH} size="lg"/>
    </span>
);

export const RightIcon = () => (
  <span>
    <FontAwesomeIcon icon={faAngleRight} size="md"/>
  </span>
);
