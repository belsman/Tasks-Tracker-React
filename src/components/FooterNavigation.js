import React from 'react';
import { NavLink } from 'react-router-dom';
import {
  AddMeasurementIcon, ListMeasurementIcon, ProgressIcon, MoreIcon,
} from './Icons';

import style from './footerNav.module.css';

const FooterNavigation = () => {
  const checkActiveLink = path => (match, location) => location.pathname === path;

  return (
    <nav className={style.nav}>
      <div className={style.navLinks}>
        <NavLink
          to="/"
          activeClassName="navActive"
          isActive={checkActiveLink('/')}
        >
          <AddMeasurementIcon />
        </NavLink>

        <NavLink
          to="/records"
          activeClassName="navActive"
          isActive={checkActiveLink('/records')}
        >
          <ListMeasurementIcon />
        </NavLink>

        <NavLink to="#">
          <ProgressIcon />
        </NavLink>

        <NavLink to="#">
          <MoreIcon />
        </NavLink>
      </div>
    </nav>
  );
};

export default FooterNavigation;
