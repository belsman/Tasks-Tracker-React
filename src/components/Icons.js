import React from 'react';
import {
  faHourglass, faPlus, faChartLine, faChartPie, faEllipsisH,
  faAngleRight, faAngleLeft, faTerminal, faRunning,
  faBookOpen, faTools, faFilm, faSignOutAlt,
} from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export const AddMeasurementIcon = () => (
  <span>
    <FontAwesomeIcon icon={faHourglass} size="lg" />
    <sup><FontAwesomeIcon icon={faPlus} size="sm" /></sup>
  </span>
);

export const ListMeasurementIcon = () => (
  <span>
    <FontAwesomeIcon icon={faChartLine} size="lg" />
  </span>
);

export const ProgressIcon = () => (
  <span>
    <FontAwesomeIcon icon={faChartPie} size="lg" />
  </span>
);

export const MoreIcon = () => (
  <span>
    <FontAwesomeIcon icon={faEllipsisH} size="lg" />
  </span>
);

export const LogOutIcon = () => (
  <span>
    <FontAwesomeIcon icon={faSignOutAlt} size="lg" />
  </span>
);

export const RightIcon = () => (
  <span>
    <FontAwesomeIcon icon={faAngleRight} size="md" />
  </span>
);

export const LeftIcon = () => (
  <span>
    <FontAwesomeIcon icon={faAngleLeft} size="md" />
  </span>
);

export const CodingIcon = () => (
  <span>
    <FontAwesomeIcon icon={faTerminal} size="lg" />
  </span>
);

export const RunningIcon = () => (
  <span>
    <FontAwesomeIcon icon={faRunning} size="lg" />
  </span>
);

export const ReadingIcon = () => (
  <span>
    <FontAwesomeIcon icon={faBookOpen} size="lg" />
  </span>
);

export const MovieIcon = () => (
  <span>
    <FontAwesomeIcon icon={faFilm} size="lg" />
  </span>
);

export const ProjectIcon = () => (
  <span>
    <FontAwesomeIcon icon={faTools} size="lg" />
  </span>
);
