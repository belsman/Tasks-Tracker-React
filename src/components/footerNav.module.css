.nav {
  position: fixed;
  width: 100%;
  bottom: 0;
  overflow: hidden;
  background-color: #212529;
}

.navLinks {
  display: flex;
  justify-content: space-between;
}

.navLinks a {
  display: block;
  flex-basis: 25%;
  text-align: center;
  color: #fff;
  padding: 0.5rem;
  text-decoration: none;
}

.navLinks a:hover {
  background-color: #42b5e8;
}
