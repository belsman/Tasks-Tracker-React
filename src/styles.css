* {
  box-sizing: border-box;
}

body {
  margin: 0;
  font-family: Arial, Helvetica, sans-serif;
  background-color: #e8e8e8;
  margin-bottom: 43px;
}

.h2 {
  font-size: large;
  font-weight: 700;
}

.navActive {
  background-color: #42b5e8;
}

.header {
  background-color: #42b5e8;
  color: #fff;
  padding: 1rem;
  text-align: center;
}
