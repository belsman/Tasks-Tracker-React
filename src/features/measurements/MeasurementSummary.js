import React from 'react';
import { format } from 'date-fns';
import { Link } from 'react-router-dom';

import { CircularProgressbar, buildStyles } from 'react-circular-progressbar';
import 'react-circular-progressbar/dist/styles.css';

import { RightIcon } from '../../Icons';
import style from './measurementsStyle.module.css';

const MeasurementSummary = ({ record, expectedTotal }) => {
  const {
    movie, project, running, reading, coding,
  } = record;
  const recordTotal = [movie, project, running, reading, coding]
    .reduce((total, num) => total + num);

  const percentage = parseInt(((expectedTotal - recordTotal) / expectedTotal) * 100);

  const color = percentage < 50 ? 'tomato' : '#5cb85c';

  return (
    <div className={style.recordSummary}>
      <div
        className={style.recordSummary2}
        style={{ display: 'flex', alignItems: 'center' }}
      >
        <div style={{ width: 50, paddingRight: 3 }}>
          <CircularProgressbar
            value={percentage}
            text=""
            styles={buildStyles({
              pathColor: `${color}`,
              trailColor: '#d6d6d6',
            })}
          />
        </div>
        <div style={{ paddingLeft: 3 }}>
          <span className={style.timestamp}>{format(new Date(record.created_at), 'MMM dd, yyyy')}</span>
          <span className={style.duration}>
            Hours:
            {recordTotal}
          </span>
        </div>
      </div>
      <span className={style.recordSummary3}>
        <span>
          <span className={style.timeGain}>
            {expectedTotal - recordTotal}
            {' '}
          </span>
          <span className={style.unit}>Hours</span>
        </span>
        <span className={style.toDetailLink}>
          <Link to={`/records/${record.id}`}><RightIcon /></Link>
        </span>
      </span>
    </div>
  );
};

export default MeasurementSummary;
