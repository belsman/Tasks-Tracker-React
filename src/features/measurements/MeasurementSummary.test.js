import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import pretty from 'pretty';
import { act } from 'react-dom/test-utils';

/* eslint-disable no-unused-vars */
import { format } from 'date-fns';
import { Link } from 'react-router-dom';

import PropTypes from 'prop-types';
import { CircularProgressbar, buildStyles } from 'react-circular-progressbar';
import 'react-circular-progressbar/dist/styles.css';
import { RightIcon } from '../../components/Icons';
import MeasurementSummary from './MeasurementSummary';

/* eslint-enable no-unused-vars */
jest.mock('react-router-dom', () => ({
  Link() {
    return (
      <div>A Dummy Link</div>
    );
  },
}));

jest.mock('date-fns', () => ({
  format() {
    return (
      'March 13th, 1962'
    );
  },
}));

jest.mock('react-circular-progressbar', () => ({
  CircularProgressbar() {
    return (
      <div>A Dummy progressbar</div>
    );
  },
  buildStyles() {},
}));

jest.mock('../../components/Icons', () => ({
  RightIcon() {
    return (
      <div>An Icon</div>
    );
  },
}));

const record = {
  id: 1,
  running: 1,
  reading: 1,
  project: 1,
  coding: 1,
  movie: 1,
  created_at: '2013-03-10T02:00:00Z',
};

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
  act(() => {
    render(<MeasurementSummary record={record} expectedTotal={11} />, container);
  });
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders with with the correct timestamp for a given the record', () => {
  const timestamp = document.querySelector('span.timestamp');
  expect(timestamp.textContent).toBe('March 13th, 1962');
});

it('renders with with the correct unit as Hours', () => {
  const unit = document.querySelector('span.unit');
  expect(unit.textContent).toBe('Hours');
});

it('renders with with the correct duration as sum of total time spent', () => {
  const duration = document.querySelector('span.duration');
  expect(duration.textContent).toBe('Hours:5');
});

it('renders with with the correct time gain as a difference of 11 and 5', () => {
  const timeGain = document.querySelector('span.timeGain');
  expect(timeGain.textContent).toBe('6 ');
});

it('should render the expected output', () => {
  act(() => {
    render(<MeasurementSummary record={record} expectedTotal={11} />, container);
  });

  expect(
    pretty(container.innerHTML),
  ).toMatchSnapshot();
});
