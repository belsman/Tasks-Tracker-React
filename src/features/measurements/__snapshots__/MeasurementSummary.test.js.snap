// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the expected output 1`] = `
"<div class=\\"recordSummary\\">
  <div class=\\"recordSummary2\\" style=\\"display: flex; align-items: center;\\">
    <div style=\\"width: 50px; padding-right: 3px;\\">
      <div>A Dummy progressbar</div>
    </div>
    <div style=\\"padding-left: 3px;\\"><span class=\\"timestamp\\">March 13th, 1962</span><span class=\\"duration\\">Hours:5</span></div>
  </div><span class=\\"recordSummary3\\"><span><span class=\\"timeGain\\">6 </span><span class=\\"unit\\">Hours</span></span><span class=\\"toDetailLink\\"><div>A Dummy Link</div></span></span>
</div>"
`;
